sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SpannerMutationCoder.java

```diff
@@ -0,0 +1,224 @@
+package org.apache.beam.sdk.coders;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.Date;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Timestamp;
+import com.google.cloud.spanner.Type;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.spanner.ValueBinder;
+import com.google.common.io.ByteArrayDataInput;
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+/**
+ * A {@link Coder} that encodes Spanner {@link Mutation} objects.
+ */
+public class SpannerMutationCoder extends AtomicCoder<Mutation> {
+
+  @JsonCreator
+  public static SpannerMutationCoder of() {
+    return INSTANCE;
+  }
+
+  /***************************/
+
+  private static final SpannerMutationCoder INSTANCE = new SpannerMutationCoder();
+
+  private SpannerMutationCoder() {}
+
+  @Override
+  public void encode(Mutation value, OutputStream outStream, Context context)
+              throws IOException, CoderException {
+
+      if (value == null) {
+          throw new CoderException("cannot encode a null Mutation");
+      }
+
+      if (Mutation.Op.DELETE == value.getOperation()) {
+          throw new UnsupportedOperationException("DELETE Mutations not supported!");
```

Comment:
- The coder doesn't seem to be a right place for this check. Maybe PTransform#validate
- Although other use cases are useful, looking at the code in SpannerIO, I think we should restrict the code to support only InsertAndUpdate for now. Because other mutation types require different error handling/retry policy.